src/plugins/data_source_management/public/components/data_source_multi_selectable/data_source_multi_selectable.tsx

```diff
@@ -43,46 +47,49 @@ export class DataSourceMultiSelectable extends React.Component<
 
   async componentDidMount() {
     this._isMounted = true;
-    getDataSourcesWithFields(this.props.savedObjectsClient, ['id', 'title', 'auth.type'])
-      .then((fetchedDataSources) => {
-        if (fetchedDataSources?.length) {
-          // all data sources are selected by default on initial page load
-          const selectedOptions: SelectedDataSourceOption[] = fetchedDataSources.map(
-            (dataSource) => ({
-              id: dataSource.id,
-              label: dataSource.attributes?.title || '',
-              checked: 'on',
-              visible: true,
-            })
-          );
+    try {
+      const defaultDataSource = this.props.uiSettings?.get('defaultDataSource', null) ?? null;
+      let selectedOptions: SelectedDataSourceOption[] = [];
+      const fetchedDataSources = await getDataSourcesWithFields(this.props.savedObjectsClient, [
+        'id',
+        'title',
+        'auth.type',
+      ]);
 
-          if (!this.props.hideLocalCluster) {
-            selectedOptions.unshift({
-              id: '',
-              label: 'Local cluster',
-              checked: 'on',
-              visible: true,
-            });
-          }
+      if (fetchedDataSources?.length) {
+        selectedOptions = fetchedDataSources.map((dataSource) => ({
+          id: dataSource.id,
+          label: dataSource.attributes?.title || '',
+          checked: 'on',
+          visible: true,
+        }));
+      }
 
-          if (!this._isMounted) return;
-          this.setState({
-            ...this.state,
-            selectedOptions,
-          });
+      if (!this.props.hideLocalCluster) {
+        selectedOptions.unshift({
+          id: '',
+          label: 'Local cluster',
+          checked: 'on',
+          visible: true,
+        });
+      }
 
-          this.props.onSelectedDataSources(
-            selectedOptions.filter((option) => option.checked === 'on')
-          );
-        }
-      })
-      .catch(() => {
-        this.props.notifications.addWarning(
-          i18n.translate('dataSource.fetchDataSourceError', {
-            defaultMessage: 'Unable to fetch existing data sources',
-          })
-        );
+      if (!this._isMounted) return;
+
+      this.setState({
+        ...this.state,
+        selectedOptions,
+        defaultDataSource,
       });
+
+      this.props.onSelectedDataSources(selectedOptions.filter((option) => option.checked === 'on'));
```

Comment:
looks like we can just send back selectedOptions since selectedOptions always have checked field being set to `on`

Reply:
Will make the change